legacy/ui/legacy/src/main/java/com/fsck/k9/ui/settings/account/OpenPgpAppSelectDialog.java

```diff
@@ -239,13 +240,15 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
             builder.setNegativeButton(com.fsck.k9.ui.base.R.string.cancel_action, new OnClickListener() {
                 @Override
                 public void onClick(DialogInterface dialogInterface, int i) {
-                    dismiss();
+                    //dismiss();
+                    dismissAllowingStateLoss();
```

Comment:
The onClick-handler should only be called when the activity is still active. So calling `dismissAllowingStateLoss()` here is not necessary. Please revert this change.